core/dbt/contracts/sql.py

```diff
@@ -32,7 +32,7 @@ class RemoteResult(VersionedSchema):
 class RemoteCompileResultMixin(RemoteResult):
     raw_code: str
     compiled_code: str
-    node: CompileResultNode
+    node: ResultNode
```

Comment:
This may have implications for `dbt-rpc` ([here](https://github.com/dbt-labs/dbt-rpc/blob/514691d7b2d6387e879df461918a0e78fe956c96/dbt_rpc/contracts/rpc.py#L261)). I don't _think_ this should have any direct implications for `dbt-server` / `lib.py`, but worth testing to be sure.